Neos.Flow/Classes/ResourceManagement/Target/FileSystemTarget.php

```diff
@@ -392,23 +397,16 @@ protected function getResourcesBaseUri()
      * Detects and returns the website's absolute base URI
      *
      * @return string The resolved resource base URI, @see getResourcesBaseUri()
-     * @throws TargetException if the baseUri can't be resolved
+     * @throws \Neos\Flow\Http\Exception
      */
     protected function detectResourcesBaseUri()
     {
         if ($this->baseUri !== '' && ($this->baseUri[0] === '/' || strpos($this->baseUri, '://') !== false)) {
             return $this->baseUri;
         }
 
-        $requestHandler = $this->bootstrap->getActiveRequestHandler();
-        if ($requestHandler instanceof HttpRequestHandlerInterface) {
-            return $requestHandler->getHttpRequest()->getAttribute(ServerRequestAttributes::BASE_URI) . $this->baseUri;
-        }
-
-        if ($this->httpBaseUri === null) {
-            throw new TargetException(sprintf('The base URI for resources could not be detected. Please specify the "Neos.Flow.http.baseUri" setting or use an absolute "baseUri" option for target "%s".', $this->name), 1438093977);
-        }
-        return $this->httpBaseUri . $this->baseUri;
+        $httpBaseUri = (string)$this->baseUriProvider->getConfiguredBaseUriOrFallbackToCurrentRequest();
```

Comment:
Ah, one more suggestion: Should we catch the `\Neos\Flow\Http\Exception` here and throw a more meaningful here like
```php
catch (HttpException $exception) {
  throw new SomeOtherException(sprintf('Failed to determine base URI for FileSystemTarget "%s", please specify the "baseUri" option', $this->name), 1567668788, $exception);
}
```
?

Reply:
I am not sure how that is better? I thought about it as well, but then think hiding an exception with another one? What's the point. The necessary action is the same and is already outlined in the first one.